nucliadb/src/nucliadb/search/search/query.py

```diff
@@ -544,6 +560,29 @@ def get_max_tokens_answer(self) -> Optional[int]:
             return self.max_tokens.answer
         return None
 
+    async def adjust_page_size(self, request: nodereader_pb2.SearchRequest, reranker: Reranker):
+        """Some rerankers want more results than the requested by the user so
+        reranking can have more choices. This function adjust the number of
+        retrieval results requested according to the reranker needs."""
+
+        if not reranker.needs_extra_results:
+            return
+
+        async with datamanagers.with_ro_transaction() as txn:
+            external_provider = await datamanagers.kb.get_external_index_provider_metadata(
```

Comment:
If you only want to know if a kb has external index provider or not, I think you should hard cache this in memory as it is not modifiable per KB

Reply:
Not sure if now is much better, as the first query for KBs with our index will use two transactions, but I guess it's fine